deployment/aws-python-storage-stack-automation/storage_stack_lifecycle.yaml

```diff
@@ -352,6 +352,202 @@ Resources:
+              - "lambda:RemovePermission"  
+            Resource: "*"
+          - Effect: "Allow"
+            Action:
+              - "secretsmanager:GetSecretValue"  
+            Resource: !Ref wsapi
+          - Effect: Allow
+            Action:
+              - "sts:AssumeRole"
+
+            Resource: !GetAtt
+              - implodestoragelambda
+              - Arn
+      Roles:
+        - !Ref implodelambdarole
+  implodestoragelambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      Runtime: python3.8
+      Timeout: 120
+      Handler: index.lambda_handler
+      Role: !GetAtt
+        - implodelambdarole
+        - Arn
+      Code:
+        ZipFile: |
+          import json
+          import os
+          import boto3
+          import urllib3
+          from botocore.exceptions import ClientError
+          http = urllib3.PoolManager()
+
+          fss_key = os.environ["WS_API"]
+          get_url = "https://cloudone.trendmicro.com/api/filestorage/external-id"
+          post_url = "https://cloudone.trendmicro.com/api/filestorage/stacks"
+          delete_url = "https://cloudone.trendmicro.com/api/filestorage/stacks/"
+
+          def lambda_handler(event, context):
+              # get secret
+              secrets = boto3.client('secretsmanager').get_secret_value(SecretId=fss_key)
+              sm_data = json.loads(secrets["SecretString"])
+              ws_key = sm_data["wsapikey"]
+
+              #gather bucket name from event
+              bucket_name = event['detail']['requestParameters']['bucketName']
+              print("S3 Bucket: " + bucket_name)
+              
+              # gather cloud one ext id
+              r = http.request(
+                  "GET",
+                  get_url,
+                  headers={
+                      "Content-Type": "application/json",
+                      "api-secret-key": ws_key,
+                      "Api-Version": "v1",
+                    },
+                )
+              ext = json.loads(r.data.decode("utf-8"))
+              ext_id = ext["externalID"]
+              print("CloudOne ExtID: " + ext_id)
+              
+              # gather aws account ID
+              account_id = event["account"]
+              print("AWS account ID: " + account_id)
+
+              #gather stack id
+              id_call = http.request('GET', post_url, headers = {'Content-Type': 'application/json', 'api-secret-key': ws_key, 'Api-Version': 'v1'})
+              id_resp = json.loads(id_call.data.decode('utf-8'))
+              for data in id_resp['stacks']:
+                  if 'storage' in data and data['storage'] is not None:
+                      if bucket_name == data['storage']:
+                          stack_name = data['name']
+                          stack_id = data['stackID']
+                          print(stack_name)
+                          print(stack_id)
+                          stack_removal(ws_key, stack_id,stack_name)
```

Comment:
If there are a lot of stacks, you have to handle pagination with `next` and `cursor`.
Ref https://cloudone.trendmicro.com/docs/file-storage-security/api-reference/tag/Stack#operation/listStacks

Reply:
thanks, I update to handle next if exists.